### PR TITLE
Sftp connector

### PR DESCRIPTION
**Pull request by winiar93**
## Summary
Added source and task class for downloading data from SFTP  server.

## Checklist

This PR:

- [X] adds new tests (if appropriate)
- [X] updates docstrings for any new functions or function arguments (if appropriate)
- [X] updates `CHANGELOG.md` with a summary of the changes

### CODE REVIEW

File: viadot/sources/sftp.py [hunk 1]
```diff
@@ -0,0 +1,215 @@
...
+                self.recursive_listdir(os.path.join(path, attr.filename), files)
+
+            else:
+                #  if the file is a file, add it to our dict
+                files[path].append(attr.filename)
+
+        self.recursive_files = files
+        return files
+
+    def process_defaultdict(self, defaultdict: any = None):
```
Comment (Rafalz13) [inline viadot/sources/sftp.py:191]: Add return type.

File: viadot/sources/sftp.py [hunk 2]
```diff
@@ -0,0 +1,215 @@
...
+        """Process defaultdict to list of files.
+
+        Args:
+            defaultdict (any): defaultdict of recursive files.
```
Comment (Rafalz13) [inline viadot/sources/sftp.py:195]: Add  `, optional` and `Defaults to None.`

File: tests/integration/test_sftp.py [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+    data = {"country": [1, 2], "sales": [3, 4]}
+    df = pd.DataFrame(data=data)
+    return df
+
+
+@pytest.fixture
+def list_of_paths():
+
+    list_of_paths = [
+        "Country__Context30##exported.tsv",
```
Comment (Rafalz13) [inline tests/integration/test_sftp.py:21]: Is `tsv` some file extension or mistake?

**Review by Rafalz13 — CHANGES_REQUESTED** [review]
I am not sure that the `..ToADLS` and `..ToSQL` flows should be in the same file.
Cosmetic changes.

File: viadot/tasks/sftp.py [hunk 4]
```diff
@@ -0,0 +1,154 @@
...
+                credentials_sftp = local_config.get("QWASI-SFTP")
+                logger.info("Loaded credentials from local source.")
+        else:
+            credentials_sftp = self.credentials
+
+        sftp = SftpConnector(credentials_sftp=credentials_sftp)
+        sftp.get_conn()
+        logger.info("Connected to SFTP server.")
+        df = sftp.to_df(file_name=from_path, sep=sep, columns=columns)
+        logger.info("Succefully downloaded file from SFTP server.")
```
[the remark this answers is not in the document]
Comment (winiar93) [reply viadot/tasks/sftp.py:77]: paramiko should handle error

File: viadot/flows/sftp_operations.py [hunk 5]
```diff
@@ -178,9 +178,9 @@ def __init__(
 
         Args:
             name (str): The name of the flow.
-            from_path (str): Path to the file in SFTP server.
-            file_name (str): File name for local file.
-            sep (str): The separator to use to read the CSV file.
+            from_path (str, optional): Path to the file in SFTP server. Defaults to False.
+            file_name (str, optional): File name for local file. Defaults to False.
```
Comment (Rafalz13) [inline viadot/flows/sftp_operations.py:182]: according to __init__()-> `Defaults to None`

**Review by Rafalz13 — APPROVED** [review]

File: viadot/flows/sftp_operations.py (hunk 5 above)
Comment (Rafalz13) [reply viadot/flows/sftp_operations.py:182]: + Add flow_toADLS to tasks/init

**Pull request merged**